dspace-xmlui/src/main/java/org/dspace/app/xmlui/aspect/versioning/VersionManager.java

```diff
@@ -83,6 +83,7 @@ public static FlowResult processCreateNewVersion(Context context, UUID itemID, S
             context.abort();
             throw new RuntimeException(ex);
         }
+        context.complete();
```

Comment:
In the event of "submit_version" (https://github.com/DSpace/DSpace/blob/dspace-6_x/dspace-xmlui/src/main/resources/aspects/Versioning/versioning.js#L130), a redirect is called to the item submission, and then cocoon.exit(). So in this case, cocoon doesn't return to getDSContext().complete() on line 108.

I agree that the other VersionManager methods should get some additional examination to ensure the context.complete() isn't necessary there, but probably the redirect to item submission is a rather exceptional case.

Perhaps it would be cleaner to move the context.complete() to versioning.js, just before the redirect?